src/main/java/com/finalmarket/demo/logic/MainService.java

```diff
@@ -0,0 +1,47 @@
+package com.finalmarket.demo.logic;
+
+
+import jakarta.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Set;
+
+@Component
+@RequiredArgsConstructor
+public class MainService implements PredefinedLogicService {
```

Comment:
I would better to not use the same interface "PredefinedLogicService " for delegates and the main service.